### ISSUE TITLE
CLEANUP:  acpi: x86: Move MADT to common code 

### ISSUE THREAD

**Issue by PatrickRudolph** [9elements/u-boot-acpi#11]
**Commit**

https://github.com/9elements/u-boot-acpi/commit/65bd0363fda743ec9155bf80022dad9327461db9

**Requested changes**

```
> Write MADT in common code and let the SoC fill out the body by
> calling acpi_fill_madt() which must be implemented at SoC level.

This should be done using the acpi_ops struct...i.e. add a new
operation which fills allows a CPU driver to fill out the CPU info
```

**Comment by PatrickRudolph** [comment]
Resolved for QEMU SBSA in cde6a7044cc9640a2237280eb2a68be7075beb2e

**Comment by PatrickRudolph** [comment]
No longer neccessary.
